File: seleniumEasyJava/src/test/java/E2E/steps/checkbox/CheckboxSteps.java
package E2E.steps.checkbox;

import E2E.runner.TestRunner;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;
import org.junit.Assert;
import org.junit.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;

import java.util.List;

public class CheckboxSteps {
    @Given("the developer is on the checkbox demo selenium easy website")
    public void theDeveloperIsOnTheCheckboxDemoSeleniumEasyWebsite() {
        TestRunner.driver.get("https://demo.seleniumeasy.com/basic-checkbox-demo.html");
    }

//    ------------------- Single Checkbox --------------------------------
    @When("the developer clicks on single check box")
    public void theDeveloperClicksOnSingleCheckBox() {
        TestRunner.checkboxPage.singleCheckbox.click();
    }

    @Then("a success message pops up on the screen")
    public void aSuccessMessagePopsUpOnTheScreen() {
       boolean message = TestRunner.explicitWait.until(ExpectedConditions.textToBePresentInElement(
                TestRunner.checkboxPage.hiddenSuccessMessage, "Success - Check box is checked"));
        Assert.assertTrue(message);
    }

//    ------------------- Multiple Checkboxes --------------------------------

    @When("the developer clicks on the check all button")
    public void theDeveloperClicksOnTheCheckAllButton() {
        TestRunner.checkboxPage.checkboxButton.click();
    }

    @Then("all the checkboxes get checked")
    public void allTheCheckboxesGetChecked() {
        int checked = 0;
        List<WebElement> checkboxList = TestRunner.driver.findElements(By.className("cb1-element"));
        for(WebElement checkbox: checkboxList){
            if(checkbox.isSelected()){
                checked++;
            }
        }
        Assert.assertEquals(checked, checkboxList.size());
    }

}

File: seleniumEasyPython/steps/checkbox_steps.py
from behave import Given, When, Then
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.wait import WebDriverWait


@Given(u'the developer is on the checkbox demo selenium easy website')
def get_selenium_checkbox_easy_website(context):
    context.driver.get("https://demo.seleniumeasy.com/basic-checkbox-demo.html")


#     ------------------- Single Checkbox --------------------------------
@When(u'the developer clicks on single check box')
def click_single_checkbox(context):
    context.checkbox.single_checkbox().click()


@Then(u'a success message pops up on the screen')
def success_message(context):
    element = WebDriverWait(context.driver, 3).until(
        expected_conditions.text_to_be_present_in_element((By.ID, "txtAge"),
                                                          "Success - Check box is checked"))
    assert element


#     ------------------- Multiple Checkboxes --------------------------------


@When(u'the developer clicks on first check box')
def click_checkbox_1(context):
    context.checkbox.checkbox_1().click()


@When(u'the developer clicks on second check box')
def click_checkbox_2(context):
    context.checkbox.checkbox_2().click()


@When(u'the developer clicks on third check box')
def click_checkbox_3(context):
    context.checkbox.checkbox_3().click()


@Then(u'the button text changes')
def button_text_change(context):
    element = WebDriverWait(context.driver, 3).until(expected_conditions.text_to_be_present_in_element((By.ID, "check1")
                                                                                                       , "Uncheck All"))
